src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -282,10 +290,16 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
     if task.trigger:
         source_triggers = task.trigger.source_triggers
         base_image_trigger = task.trigger.base_image_trigger
-        if commit_trigger_enabled or source_triggers is not None:
-            status = None
-            if commit_trigger_enabled is not None:
-                status = TriggerStatus.enabled.value if commit_trigger_enabled else TriggerStatus.disabled.value
+        if (commit_trigger_enabled or pull_request_trigger_enabled) or source_triggers is not None:
+            #using a set to prevent duplicate event types per trigger
+            source_trigger_events = set()
```

Comment:
The set is used in "update" because the CLI may be updating an existing trigger event list. In the case where the user wants to enable an event that is already in the list, we want to make sure that we are not adding an event twice. The set will ensure uniqueness. However, we do not have this problem in "create" so we can just use a list.